src/Channel.cpp

```diff
@@ -81,8 +81,12 @@ JointState Channel::getJointState() const {
 
     switch (m_control_mode) {
         case CONTROL_NONE:
-        case CONTROL_OPEN_LOOP:
+        case CONTROL_OPEN_LOOP: {
+            int32_t feedback = get<EncoderCounter>();
+            state.position = m_factors.positionToSI(feedback,
+                                                    m_joint_state_position_source);
```

Comment:
This is valid only if m_joint_state_position_source is encoder. Otherwise, nothing should be done at all. Add the if guard *and* call the right factor conversion method directly.

This code will break systems that don't have encoders (since get<EncoderCounter> will throw) and we would not have any way to disable it.